src/App.tsx

```diff
@@ -59,18 +59,88 @@ function App() {
   const [selectedAccounts, setSelectedAccounts] = useState<
     Record<string, `${string}:${string}:${string}`>
   >({});
+  const [walletSessionChangedHistory, setWalletSessionChangedHistory] =
+    useState<{ timestamp: number; data: any }[]>([]);
+
+  const handleSessionChangedNotification = (notification: any) => {
+    setWalletSessionChangedHistory((prev) => [
+      { timestamp: Date.now(), data: notification },
+      ...prev,
+    ]);
```

Comment:
doesn't really matter and I know you're doing this so newest shows up first in UI, but it feels wrong to prepend haha

Reply:
yeah 🤷...